### ISSUE TITLE
Standalone comments should be preserved in variable table

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#1088]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049) by masiar.m... on 27 Aug 2012*

I would like to propose that you to add possibility in RIDE 1.0 to disable reformatting of comments in TSV files when saving them.

To give an example: 
I usually like to have comments in the *Variable* table of my TSV-formatted test case file (e.g. for grouping them, or for commenting out unused lines).
The problem is that RIDE will reformat the file when making a change in the TSV file with RIDE and saving it, such that all line breaks between consecutive comment cells will be removed and the comments will be all put in a single line (or multiple lines with line continuation) and only separated by the cell borders. 

Please see the attached email for an example.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c1) by masiar.m... on 27 Aug 2012*


... removed email attachment and replaced by pdf ...

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c2) by @jussimalinen on 28 Aug 2012*


<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=1052" title="Ride does not preserve standalone comments in variable table"> Issue 1052 </a> has been merged into this issue.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c3) by @jussimalinen on 28 Aug 2012*


This is blocked by <a href="https://code.google.com/p/robotframework/issues/detail?id=1210" rel="nofollow">https://code.google.com/p/robotframework/issues/detail?id=1210</a> in Robot Framework

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c4) by @pekkaklarck on 28 Aug 2012*


I don't think this needs to be optional, RIDE simply shouldn't reformat standalone comments in variable table this way. Additionally, this doesn't affect only TSV but also all other test data formats.

As Jussi already commented, the actual fix needs to be done on Robot's side. Luckily that seems to be relatively straightforward and I thus scope this issue already to the next RIDE release.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c5) by @pekkaklarck on 29 Aug 2012*


Robot's parsing and writer modules have now been enhanced to handle standalone comments in variable table. We already tested that this back-end code works fine in RIDE, but some more work still needed in RIDE's UI to support this new functionality better.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c6) by @mkorpela on 30 Aug 2012*


The tree view does not at all show the comments now.

<a href="https://github.com/robotframework/RIDE/commit/8798bb22031d130cb6ec7112c6f53c561c9f4b8b" rel="nofollow">https://github.com/robotframework/RIDE/commit/8798bb22031d130cb6ec7112c6f53c561c9f4b8b</a>

have to still test that other functionalities work ok.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1049#c7) by @mkorpela on 30 Aug 2012*


Preliminary testing ok.

Marking as Done.
